bindings/python/dlite-collection-python.i

```diff
@@ -295,26 +312,54 @@ class Collection(Instance):
         """Returns a generator over all labels."""
         return self.get_subjects(p='_is-a', o='Instance')
 
-    def get_relations(self, s=None, p=None, o=None, rettype='T'):
+    def get_relations(self, s=None, p=None, o=None, d=None, rettype='t'):
         """Returns a generator over all relations matching the given
-        values of `s`, `p` and `o`."""
-        return _CollectionIter(self, s=s, p=p, o=o, rettype=rettype)
+        values of `s` (subject), `p` (predicate), `o` (object) and `d`
+        (datatype). See docstring for dlite.Collection for more info.
+
+        The following values for `rettype` are supported:
+        - 'I': Iterate over instances.
+        - 'R': Iterate over relation objects.
+        - 'T': Iterate over (s,p,o,d) tuples.
+        - 't': Iterate over (s,p,o) tuples.
+        - 's': Iterate over subjects.
+        - 'p': Iterate over predicates.
+        - 'o': Iterate over objects.
+        - 'd': Iterate over datatypes.
+        """
+        return _CollectionIter(self, s=s, p=p, o=o, d=d, rettype=rettype)
 
-    def get_subjects(self, p=None, o=None):
+    def get_subjects(self, p=None, o=None, d=None):
         """Returns a generator over all subjects of relations matching the
-        given values of `p` and `o`."""
-        return _CollectionIter(self, s=None, p=p, o=o, rettype='s')
+        given values of `p`, `o` and `d`."""
+        return _CollectionIter(self, s=None, p=p, o=o, d=d, rettype='s')
 
-    def get_predicates(self, s=None, o=None):
+    def get_predicates(self, s=None, o=None, d=None):
         """Returns a generator over all predicates of relations matching the
-        given values of `s` and `o`."""
-        return _CollectionIter(self, s=s, p=None, o=o, rettype='p')
+        given values of `s`, `o` and `d`."""
+        return _CollectionIter(self, s=s, p=None, o=o, d=d, rettype='p')
 
-    def get_objects(self, s=None, p=None):
+    def get_objects(self, s=None, p=None, d=None):
         """Returns a generator over all subjects of relations matching the
-        given values of `s` and `p`."""
-        return _CollectionIter(self, s=s, p=p, o=None, rettype='o')
+        given values of `s`, `p` and `d`."""
+        return _CollectionIter(self, s=s, p=p, o=None, d=d, rettype='o')
+
+    def value(self, s=None, p=None, o=None, d=None, default=None, any=False):
+        """Return pointer to the value for a pair of two criteria.
 
+        Useful if one knows that there may only be one value.
+
+        Parameters:
+            s, p, o: Criteria to match. Two of these must be non-NULL.
+            d: If not NULL, the required datatype of literal objects.
+            default: Value to return if no matches are found.
```

Comment:
A bit confusing to start referring to NULL, as this is in python. I think it is beter to use None and if it is importatnt to explain the connection to C, do that.

Reply:
Well spotted. I changed it to Python nomenclouture.